src/BSON/Binary.stub.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/** @generate-function-entries */
+
+namespace MongoDB\BSON
+{
+    final class Binary
+    {
+        public final function __construct(string $data, int $type) {}
+
+        final public function getData(): string {}
+
+        final public function getType(): int {}
+
+        public static function __set_state(array $properties): object {}
```

Comment:
Unfortunately, `self` isn't possible:
> $ ./build/gen_stub.php 
In ./src/BSON/Binary.stub.php:
MongoDB\BSON\Binary::__set_state(): The exact class name must be used instead of "self"

However, using `Binary` works, so I'll be changing those.

Reply:
The prohibition on `self` dates back to https://github.com/php/php-src/commit/80e5ad5a293117552cf1d616b3a7e4b23dfc8fa5 and https://github.com/php/php-src/pull/7215. That PR references https://github.com/php/doc-en/pull/624, although I didn't find any outright discussion about why `self` is prohibited there (just that they opted to use class names instead of `self` for docs).

It looks like `static` might work if you'd rather use that; otherwise, no objections to just hard-coding the specific class name for these methods.